src/Layout.c

```diff
@@ -770,9 +777,112 @@ static void layoutNodeImpl(css_node_t *node, float parentMaxWidth) {
+  // http://www.w3.org/TR/2012/CR-css3-flexbox-20120918/#layout-algorithm
+  // section 9.4
+  //
+  if (linesCount > 1 &&
+      (!isUndefined(node->layout.dimensions[dim[crossAxis]])))
+  {
+    float nodeCrossAxisInnerSize = node->layout.dimensions[dim[crossAxis]] -
+        getPaddingAndBorderAxis(node, crossAxis);
+    float remainingCrossDim = nodeCrossAxisInnerSize - linesCrossDim;
+
+    float crossDimAdd = 0;
+    float currentLead = getPaddingAndBorder(node, leading[crossAxis]);
+
+    css_align_t alignContent = getAlignContent(node);
+    if (alignContent == CSS_ALIGN_FLEX_END) {
+      currentLead += remainingCrossDim;
+    }
+    else if (alignContent == CSS_ALIGN_CENTER) {
+      currentLead += remainingCrossDim / 2;
+    }
+    else if (alignContent == CSS_ALIGN_STRETCH) {
+      if (nodeCrossAxisInnerSize > linesCrossDim) {
+        crossDimAdd = (remainingCrossDim / linesCount);
+      }
+    }
+
+    // find the first node on the first line
+    for (i = 0; i < node->children_count; ) {
+      int startIndex = i;
+      int lineIndex = -1;
+
+      // get the first child on the current line
+      {
+        child = node->get_child(node->context, i);
+        if (getPositionType(child) != CSS_POSITION_RELATIVE) {
+          ++i;
+          continue;
+        }
+        lineIndex = child->line_index;
+      }
+
+      // compute the line's height and find the endIndex
+      float lineHeight = 0;
+      for (ii = startIndex; ii < node->children_count; ++ii) {
+        child = node->get_child(node->context, ii);
+        if (getPositionType(child) != CSS_POSITION_RELATIVE) {
+          continue;
+        }
+        if (child->line_index != lineIndex) {
+          break;
+        }
+        if (!isUndefined(child->layout.dimensions[dim[crossAxis]])) {
+          lineHeight = fmaxf(lineHeight,child->layout.dimensions[dim[crossAxis]] +
+                             getMarginAxis(child,crossAxis));
+        }
+      }
+      int endIndex = ii;
+      lineHeight += crossDimAdd;
+
+      for (ii = startIndex; ii < endIndex; ++ii) {
+        child = node->get_child(node->context, ii);
+        if (getPositionType(child) != CSS_POSITION_RELATIVE) {
+          continue;
+        }
+
+        css_align_t alignItem = getAlignItem(node, child);
+        float crossPosition = child->layout.position[pos[crossAxis]]; // preserve current position if someting goes wrong with alignItem?
+        if (alignItem == CSS_ALIGN_FLEX_START) {
+          crossPosition = currentLead + getMargin(child,leading[crossAxis]);
+        }
+        else if (alignItem == CSS_ALIGN_FLEX_END) {
+          crossPosition = currentLead + lineHeight -
+            getMargin(child,trailing[crossAxis]) -
+            child->layout.dimensions[dim[crossAxis]];
+        }
+        else if (alignItem == CSS_ALIGN_CENTER) {
+          float childHeight = child->layout.dimensions[dim[crossAxis]];
+          crossPosition = currentLead + ((lineHeight - childHeight)/2);
```

Comment:
nit: you don't need parenthesis around (a / 2) because / has precedence over +
nit: space around /